internal/pkg/consensus/election.go

```diff
@@ -231,8 +231,8 @@ func filterSectorInfosByIndex(allSectorInfos []abi.SectorInfo, challengeIDs []ui
 	}
 
 	var filteredSectorInfos []abi.SectorInfo
-	for _, si := range allSectorInfos {
-		if _, ok := idSet[uint64(si.SectorNumber)]; ok {
+	for i, si := range allSectorInfos {
+		if _, ok := idSet[uint64(i)]; ok {
```

Comment:
This is what I know:
1. The values we get back are called challenge indexes, not IDs or numbers.
2. The sectors numbers we have from the lotus seed start at 3, not 0 or 1. Granted those sectors are old now.
3. The challenge index we get in the test is 1, which causes an error when we try to find it by sector num.
4. When we look the sector up by index, proof generation works and it validates in the miner actor.

This all leads me to believe that sector numbers don't have to be sequential, and that we should be looking these things up by index instead of sector num, but I don't have any protocol to back this up.

Reply:
The challenge ID is an index in the ordered set of sector numbers. Note that the bitfield type now has a `Slice` operation that lets you index it.